packages/flame/lib/src/experimental/geometry/shapes/polygon.dart

```diff
@@ -241,4 +243,58 @@ class Polygon extends Shape {
 
   @override
   String toString() => 'Polygon($vertices)';
+
+  @override
+  Vector2 randomPoint({Random? random, bool within = true}) {
+    final randomGenerator = random ?? randomFallback;
+    if (within) {
+      final result = Vector2.zero();
+      final min = aabb.min;
+      final max = aabb.max;
+
+      while (true) {
+        final randomX = min.x + randomGenerator.nextDouble() * (max.x - min.x);
+        final randomY = min.y + randomGenerator.nextDouble() * (max.y - min.y);
+        result.setValues(randomX, randomY);
+
+        if (containsPoint(result)) {
+          return result;
+        }
+      }
+    } else {
+      return Polygon.randomPointAlongEdges(_vertices, random: randomGenerator);
+    }
+  }
+
+  /// Returns a random point on the [vertices].
+  static Vector2 randomPointAlongEdges(
+    List<Vector2> vertices, {
+    Random? random,
+  }) {
+    final randomGenerator = random ?? randomFallback;
+    final verticesLengths2 = <double>[];
+    var totalLength = 0.0;
+    for (final (i, startPoint) in vertices.indexed) {
+      final endPoint = vertices[(i + 1) % vertices.length];
+      final length2 = pow(endPoint.x - startPoint.x, 2).toDouble() +
+          pow(endPoint.y - startPoint.y, 2).toDouble();
+      verticesLengths2.add(length2);
+      totalLength += length2;
+    }
+    final pointOnEdges = randomGenerator.nextDouble() * totalLength;
+    var vertexIndex = 0;
+    var currentEndPoint = 0.0;
+    while (vertexIndex < verticesLengths2.length) {
+      currentEndPoint += verticesLengths2[vertexIndex];
+      if (currentEndPoint >= pointOnEdges) {
+        break;
+      }
+      vertexIndex++;
+    }
+    final startPoint = vertices[vertexIndex];
+    final endPoint = vertices[(vertexIndex + 1) % vertices.length];
+    final t = randomGenerator.nextDouble();
```

Comment:
you don't need to generate another random double. just decrease pointOnEdges as you iterate the edges and t is the "remainder" for this edge

Reply:
Ah, good idea!

Reply:
Hmm, then we'll have to do a square root though, I think it probably is cheaper to generate another double?